implementations/rust/ockam/ockam_api/src/cli_state.rs

```diff
@@ -264,6 +264,39 @@ impl IdentitiesState {
         Ok(IdentityState { path, config })
     }
 
+    pub fn list(&self) -> Result<Vec<IdentityState>> {
+        let mut identities = vec![];
+        for entry in std::fs::read_dir(&self.dir)? {
+            let entry = entry?;
+            let filename = entry.file_name().into_string()
+                .map_err(|_| CliStateError::Io(std::io::Error::new(
+                std::io::ErrorKind::InvalidData,
+                "node's directory has an invalid name",
+            )))?;
+            if !entry.file_type()?.is_dir() && filename.ends_with(".json") {
+                let identity_name = &filename[0..filename.len()-5];
+                identities.push(self.get(identity_name)?);
+            }
+        }
+        Ok(identities)
+    }
+
+    pub fn delete(&self, name: &str, delete_symlink: bool) -> Result<()> {
+        let path = {
+            let mut path = self.dir.clone();
+            path.push(format!("{}.json", name));
+            if !path.exists() {
+                return Err(CliStateError::NotFound(format!("identity `{name}`")));
+            }
+            path
+        };
+        std::fs::remove_file(path)?;
+        if delete_symlink {
+            // TODO Delete the symlink
```

Comment:
You can check how to do this [here](https://github.com/build-trust/ockam/blob/4036a7cdfe0de3e59ccb0cd6fd8a58970c9643e9/implementations/rust/ockam/ockam_api/src/cli_state.rs#L428).